### PR TITLE
fix!: Strict TypeScript with react-hook-form@6

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- ~[ ] Related to Abstract designs:~
- [x] Related to JIRA ticket: WEB-973
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

Enables `strict: true` to definitively enable `noImplicitAny` and other strict TypeScript compiler settings all around client-modules. Yay!!

A few small but technically breaking changes here and there. Leaving in draft for a bit so we can test them out. Turns out `react-hook-form` renamed a bunch of its core types in v6.

Relies on https://github.com/react-hook-form/react-hook-form/issues/1963 to improve `react-hook-form`'s typings around nullable refs.

Corresponding monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/18530

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "noUnusedLocals": true,
     "noUnusedParameters": false,
     "sourceMap": true,
+    "strict": true,
```
Comment (jakemhiller) [inline tsconfig.json:14]: yassss

File: packages/gamut/src/Tabs/TabList/index.tsx [hunk 2]
```diff
@@ -31,9 +31,9 @@ export const TabList: FunctionComponent<TabListProps> = ({
   const classes = cx(s.tabList, className, { [s.center]: center });
   return (
     <div className={classes} role="tablist" style={{ maxWidth }}>
-      {React.Children.toArray(children)
+      {(React.Children.toArray(children) as any)
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Tabs/TabList/index.tsx:34]: I gave up. @ me if you can get it to work.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.42-alpha.cd92f9.0<br/>@codecademy/gamut@10.0.0-alpha.cd92f9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f10b2a3355f4301d250cab3--codecademy-styleguide.netlify.app>https://5f10b2a3355f4301d250cab3--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f10b2a3355f4301d250cab3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@0.3.4-alpha.6b77b9.0<br/>@codecademy/gamut-labs@1.7.42-alpha.6b77b9.0<br/>@codecademy/gamut@10.0.0-alpha.6b77b9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f10b94be6228d75dc5afb76--codecademy-styleguide.netlify.app>https://5f10b94be6228d75dc5afb76--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f10b94be6228d75dc5afb76>Deploy Logs</a></p>

**Comment by JoshuaKGoldberg** [comment]
@jakemhiller I still have to validate react-hook-form@6 on the monolith, but it generally looks good here. Just a heads up!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@0.3.4-alpha.3eb1ce.0<br/>@codecademy/gamut-labs@1.7.42-alpha.3eb1ce.0<br/>@codecademy/gamut@10.0.0-alpha.3eb1ce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f11de6d637fe57c43c0e02a--codecademy-styleguide.netlify.app>https://5f11de6d637fe57c43c0e02a--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f11de6d637fe57c43c0e02a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f11e02c7ad6e101ec793e56--codecademy-styleguide.netlify.app>https://5f11e02c7ad6e101ec793e56--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f11e02c7ad6e101ec793e56>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@0.3.4-alpha.0f96cc.0<br/>@codecademy/gamut-labs@1.7.42-alpha.0f96cc.0<br/>@codecademy/gamut@10.0.0-alpha.0f96cc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f120ab5a7d0d4d292f8fb56--codecademy-styleguide.netlify.app>https://5f120ab5a7d0d4d292f8fb56--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f120ab5a7d0d4d292f8fb56>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@0.3.4-alpha.111d40.0<br/>@codecademy/gamut-labs@1.7.46-alpha.111d40.0<br/>@codecademy/gamut@10.0.0-alpha.111d40.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f15f562a0a29e0259ec066b--codecademy-styleguide.netlify.app>https://5f15f562a0a29e0259ec066b--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f15f562a0a29e0259ec066b>Deploy Logs</a></p>

**Pull request merged**